docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+# How to Work with Money
+
+Money is a very important concept for every e-commerce project.
+In Shopsys Framework, all monetary values (*prices, account balances, discount amounts, price limits etc.*) are represented by an instance of [the `Money` class](/packages/framework/src/Component/Money/Money.php).
+
+This approach has several advantages:
+- it avoids problems with floating point number calculations and comparisons (see [official PHP documentation](http://php.net/manual/en/language.types.float.php) for details)
+- allows easy-to-use interfaces with consistent type-hinting so you can be sure what type of value you should be using
+- prevents accidental conversion to unexpected types (which may be problematic eg. when using the `===` operator)
+- makes the application design clearer and future changes easier
+
+**Table of Contents:**
+- [Money class](#money-class)
+- [Money in Forms](#money-in-forms)
+- [Money in Twig Templates](#money-in-twig-templates)
+- [Money in Javascript](#money-in-javascript)
+- [Money in Doctrine](#money-in-doctrine)
+- [Unit and Functional Tests](#unit-and-functional-tests)
+- [Price class](#price-class)
+
+## Money class
+
+`Money` is an immutable [value object](https://codete.com/blog/value-objects/).
+
+It uses a decimal representation of the money amount and it does not contain any reference to the used currency.
+You can get the decimal representation as a `string` via the `getAmount` method.
+
+*Note: If in doubt about the results of any method, you can take a look at [its unit tests](/packages/framework/tests/Unit/Component/Money/MoneyTest.php) which contain many examples of the class' behavior.*
+
+### Construction
+
+`Money` can be constructed directly from integers and numeric strings as they are able to represent decimal numbers precisely: `$tenDollars = Money::create(10)`, `$oneAndHalfEuro = Money::create('1.20')`.
```

Comment:
I did want to communicate that $10 may be represented by `Money::create(10)` instead of `Money::create(1000)` (sometimes money is created by the smallest subunit - eg. cents - and I wanted to clear that out with an example).

You're definitely onto something with that `1.5 != 1.2` issue you mention though :smile: